File: binance4j-core/src/main/java/com/binance4j/core/dto/Signature.java
package com.binance4j.core.dto;

/** The request signature. */
public enum Signature {
	/** Endpoint can be accessed freely. */
	NONE,
	/** Endpoint requires sending a valid API-Key and signature. */
	TRADE,
	/** Endpoint requires sending a valid API-Key and signature. */
	MARGIN,
	/** Endpoint requires sending a valid API-Key and signature. */
	USER_DATA,
	/** Endpoint requires sending a valid API-Key. */
	USER_STREAM,
	/** Endpoint requires sending a valid API-Key. */
	MARKET_DATA,
}

File: src/main/kotlin/com/binance4j/connectors/core/dto/Signature.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.connectors.core.dto

/** The request signature.  */
enum class Signature {
    /** Endpoint can be accessed freely.  */
    NONE,

    /** Endpoint requires sending a valid API-Key and signature.  */
    TRADE,

    /** Endpoint requires sending a valid API-Key and signature.  */
    MARGIN,

    /** Endpoint requires sending a valid API-Key and signature.  */
    USER_DATA,

    /** Endpoint requires sending a valid API-Key.  */
    USER_STREAM,

    /** Endpoint requires sending a valid API-Key.  */
    MARKET_DATA
}